source/common/conn_pool/conn_pool_base.cc

```diff
@@ -359,7 +354,12 @@ void ConnPoolImplBase::closeIdleConnectionsForDrainingPool() {
   }
 }
 
-void ConnPoolImplBase::drainConnectionsImpl() {
+void ConnPoolImplBase::drainConnectionsImpl(DrainBehavior drain_behavior) {
+  if (drain_behavior == Envoy::ConnectionPool::DrainBehavior::DrainAndDelete) {
+    is_draining_ = true;
```

Comment:
I thought we wanted to add more asserts that no new streams are created if is_draining_ is true? Am I missing that?

Reply:
Ah yes, good point! I've added an ASSERT in ConnPoolImplBase::newStreamImpl. It did make me wonder if we should rename is_draining_ to is_draining_for_deletion_ to match the name of the enum (and sine it's not set on all drain behaviors, just one of them).

Reply:
Sure renaming SGTM (or just store the enum itself somehow)

Reply:
Renamed. I notice this matches the description in the .h file nicely now:

```
  // Whether the connection pool is currently in the process of closing                                                                                                                                                     
  // all connections so that it can be gracefully deleted.                                                                                                                                                                  
```
It's like we planned it or something :)